File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/largepicture/LargeImageListAdapter.java
package com.didichuxing.doraemonkit.kit.largepicture;

import android.content.Context;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.blankj.utilcode.util.ConvertUtils;
import com.blankj.utilcode.util.ToastUtils;
import com.didichuxing.doraemonkit.DoraemonKit;
import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.picasso.MemoryPolicy;
import com.didichuxing.doraemonkit.ui.widget.recyclerview.AbsRecyclerAdapter;
import com.didichuxing.doraemonkit.ui.widget.recyclerview.AbsViewBinder;
import com.didichuxing.doraemonkit.util.ClipboardUtils;
import com.didichuxing.doraemonkit.picasso.DokitPicasso;

import java.text.DecimalFormat;

public class LargeImageListAdapter extends AbsRecyclerAdapter<AbsViewBinder<LargeImageInfo>, LargeImageInfo> {
    private DecimalFormat mDecimalFormat = new DecimalFormat("#0.00");

    public LargeImageListAdapter(Context context) {
        super(context);
    }

    @Override
    protected AbsViewBinder<LargeImageInfo> createViewHolder(View view, int viewType) {
        return new ItemViewHolder(view);
    }

    @Override
    protected View createView(LayoutInflater inflater, ViewGroup parent, int viewType) {
        return inflater.inflate(R.layout.dk_item_large_img_list, parent, false);
    }

    private class ItemViewHolder extends AbsViewBinder<LargeImageInfo> {
        private ImageView iv;
        private TextView tvLink;
        /**
         * 框架
         */
        private TextView tvFrameWork;

        /**
         * 文件大小
         */
        private TextView tvFileSize;
        /**
         * 内存大小
         */
        private TextView tvMemorySize;
        /**
         * 图片尺寸
         */
        private TextView tvSize;

        private Button btnCopy;

        public ItemViewHolder(View view) {
            super(view);
        }

        @Override
        protected void getViews() {
            iv = getView(R.id.iv);
            tvLink = getView(R.id.tv_link);
            tvFrameWork = getView(R.id.tv_framework);
            tvFileSize = getView(R.id.tv_file_size);
            tvMemorySize = getView(R.id.tv_memory_size);
            tvSize = getView(R.id.tv_size);
            btnCopy = getView(R.id.btn_copy);
        }

        @Override
        public void bind(final LargeImageInfo largeImageInfo) {
            DokitPicasso.with(DoraemonKit.APPLICATION)
                    .load(largeImageInfo.getUrl())
                    .memoryPolicy(MemoryPolicy.NO_CACHE)
                    .resize(ConvertUtils.dp2px(100), ConvertUtils.dp2px(100))
                    .centerCrop()
                    .into(iv);
            tvLink.setText(largeImageInfo.getUrl());
            tvFrameWork.setText(String.format("framework:%s", largeImageInfo.getFramework()));
            tvFileSize.setText(String.format("fileSize:%s", mDecimalFormat.format(largeImageInfo.getFileSize()) + "KB"));
            tvMemorySize.setText(String.format("memorySize:%s", mDecimalFormat.format(largeImageInfo.getMemorySize()) + "MB"));
            tvSize.setText(String.format("width:%s   height:%s", largeImageInfo.getWidth(), largeImageInfo.getHeight()));
            btnCopy.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    ClipboardUtils.copyUri(Uri.parse(largeImageInfo.getUrl()));
                    ToastUtils.showShort("image url  has copied");
                }

            });
        }


    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/largepicture/LargeImageListAdapter.kt
package com.didichuxing.doraemonkit.kit.largepicture

import android.net.Uri
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.blankj.utilcode.util.ConvertUtils
import com.blankj.utilcode.util.ToastUtils
import com.didichuxing.doraemonkit.DoraemonKit
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.picasso.DokitPicasso
import com.didichuxing.doraemonkit.picasso.MemoryPolicy
import com.didichuxing.doraemonkit.util.ClipboardUtils
import com.didichuxing.doraemonkit.widget.bravh.BaseQuickAdapter
import com.didichuxing.doraemonkit.widget.bravh.viewholder.BaseViewHolder
import java.text.DecimalFormat

/**
 * @author:  maple
 * @time:  2020/6/8 - 15:40
 * @desc: 大图列表
 */
class LargeImageListAdapter( ) : BaseQuickAdapter<LargeImageInfo, BaseViewHolder>(R.layout.dk_item_large_img_list, mutableListOf<LargeImageInfo>()) {
    private val mDecimalFormat = DecimalFormat("#0.00")
    override fun convert(holder: BaseViewHolder, largeImageInfo: LargeImageInfo) {
        try {
            val resourceUrl = largeImageInfo.getUrl()?.toInt() ?: throw RuntimeException("url 转换异常")
            DokitPicasso.with(DoraemonKit.APPLICATION)
                    .load(resourceUrl)
                    .memoryPolicy(MemoryPolicy.NO_CACHE)
                    .resize(ConvertUtils.dp2px(100F), ConvertUtils.dp2px(100F))
                    .centerCrop()
                    .into(holder.getView<ImageView>(R.id.iv))
        } catch (e: Exception) {
            DokitPicasso.with(DoraemonKit.APPLICATION)
                    .load(largeImageInfo.getUrl())
                    .memoryPolicy(MemoryPolicy.NO_CACHE)
                    .resize(ConvertUtils.dp2px(100F), ConvertUtils.dp2px(100F))
                    .centerCrop()
                    .into(holder.getView<ImageView>(R.id.iv))
        }
        if (largeImageInfo.getMemorySize() == 0.toDouble()) {
            holder.getView<TextView>(R.id.tv_framework).text = String.format("framework:%s", "network")
            holder.getView<TextView>(R.id.tv_memory_size).visibility = View.GONE
            holder.getView<TextView>(R.id.tv_size).visibility = View.GONE
        } else {
            holder.getView<TextView>(R.id.tv_framework).text = String.format("framework:%s", largeImageInfo.getFramework())
            holder.getView<TextView>(R.id.tv_memory_size).visibility = View.VISIBLE
            holder.getView<TextView>(R.id.tv_size).visibility = View.VISIBLE
        }
        if (largeImageInfo.getFileSize() == 0.toDouble()) {
            holder.getView<TextView>(R.id.tv_file_size).visibility = View.GONE
        } else {
            holder.getView<TextView>(R.id.tv_file_size).visibility = View.VISIBLE
        }
        holder.getView<TextView>(R.id.tv_file_size).text = String.format("fileSize:%s", mDecimalFormat.format(largeImageInfo.getFileSize()) + "KB")
        holder.getView<TextView>(R.id.tv_memory_size).text = String.format("memorySize:%s", mDecimalFormat.format(largeImageInfo.getMemorySize()) + "MB")
        holder.getView<TextView>(R.id.tv_size).text = String.format("width:%s   height:%s", largeImageInfo.getWidth(), largeImageInfo.getHeight())
        holder.getView<Button>(R.id.btn_copy).setOnClickListener(View.OnClickListener {
            ClipboardUtils.copyUri(Uri.parse(largeImageInfo.getUrl()))
            ToastUtils.showShort("image url  has copied")
        })
    }
}